interpreter/cling/lib/Interpreter/Interpreter.cpp

```diff
@@ -527,17 +527,23 @@ namespace cling {
   Interpreter::CompilationResult
   Interpreter::process(const std::string& input, Value* V /* = 0 */,
                        Transaction** T /* = 0 */) {
-    CompilationOptions CO;
-    CO.DeclarationExtraction = 1;
-    CO.ValuePrinting = CompilationOptions::VPAuto;
-    CO.ResultEvaluation = (bool)V;
-    CO.DynamicScoping = isDynamicLookupEnabled();
-    CO.Debug = isPrintingDebug();
-    CO.CheckPointerValidity = 1;
-
+    CompilationOptions declareCO;
```

Comment:
Sure!
________________________________
From: Axel Naumann [notifications@github.com]
Sent: 25 April 2016 16:07
To: karies/root
Cc: Bianca-Cristina Cristescu; Author
Subject: Re: [karies/root] Correct CO for DeclareInternal in cling:Interpreter::process. (#175)


In interpreter/cling/lib/Interpreter/Interpreter.cpp<https://github.com/karies/root/pull/175#discussion_r60918215>:

> @@ -527,17 +527,23 @@ namespace cling {
>    Interpreter::CompilationResult
>    Interpreter::process(const std::string& input, Value* V /* = 0 */,
>                         Transaction** T /* = 0 */) {
> -    CompilationOptions CO;
> -    CO.DeclarationExtraction = 1;
> -    CO.ValuePrinting = CompilationOptions::VPAuto;
> -    CO.ResultEvaluation = (bool)V;
> -    CO.DynamicScoping = isDynamicLookupEnabled();
> -    CO.Debug = isPrintingDebug();
> -    CO.CheckPointerValidity = 1;
> -
> +    CompilationOptions declareCO;


Can we keep them local to the if below?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/karies/root/pull/175/files/0d62758965bde7b6aee950468d38c02387d58439#r60918215>